spec/models/case_log_spec.rb

```diff
@@ -1087,6 +1087,64 @@ def check_rsnvac_referral_validation(referral)
       end
     end
 
+    context "when saving previous address" do
+      before do
+        stub_request(:get, /api.postcodes.io/)
+          .to_return(status: 200, body: "{\"status\":200,\"result\":{\"admin_district\":\"Manchester\"}}", headers: {})
+      end
+
+      let!(:address_case_log) do
+        described_class.create({
+          managing_organisation: organisation,
+          owning_organisation: organisation,
+          previous_postcode_known: "Yes",
+          previous_postcode: "M1 1AE",
+        })
+      end
+
+      it "correctly infers prevloc" do
+        record_from_db = ActiveRecord::Base.connection.execute("select prevloc from case_logs where id=#{address_case_log.id}").to_a[0]
+        expect(address_case_log.prevloc).to eq("Manchester")
+        expect(record_from_db["prevloc"]).to eq("E08000003")
+      end
+
+      it "errors if the previous postcode is emptied" do
```

Comment:
This seems wrong? What if you realise you input an incorrect one and actually don't know it?

Reply:
hmm, this is how we're doing it for the current postcode. I think the point is that `No` should be selected for `postcode known`

Reply:
Ah I see, makes sense